### ISSUE TITLE
edge-19.4.2 isn't upgrading the proxy containers

### ISSUE THREAD

**Issue by jon-walton** [linkerd/linkerd2#2659]
### What is the issue?

Starting off with a cluster running edge-19.4.1 and running the command

`l upgrade > install.yaml`

results in ```"version":"edge-19.4.1"``` remaining in the ```linkerd-config``` configmap ```global``` key

as well as ```linkerd.io/proxy-version: edge-19.4.1``` staying within all control plane deployment annotations and ```image: gcr.io/linkerd-io/proxy:edge-19.4.1``` as the proxy image

```k -n ingress get deploy -o yaml nginx-ingress-internal-controller | l inject - | less```

also results in the following pod annotations:

```
      annotations:
        linkerd.io/created-by: linkerd/cli edge-19.4.2
        linkerd.io/identity-mode: default
        linkerd.io/inject: enabled
        linkerd.io/proxy-version: edge-19.4.1
```

and proxy image being used

```gcr.io/linkerd-io/proxy-init:edge-19.4.1```

### Logs, error output, etc

(If the output is long, please create a [gist](https://gist.github.com/) and
paste the link here.)

#### `linkerd check` output

```text
l check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ control plane namespace exists
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

Status check results are √
```

```text
l check --proxy -n linkerd
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ control plane namespace exists
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

linkerd-data-plane
------------------
√ data plane namespace exists
√ data plane proxies are ready
√ data plane proxy metrics are present in Prometheus
‼ data plane is up-to-date
    linkerd/linkerd-prometheus-775b5b9c67-dt5h9: is running version 19.4.1 but the latest edge version is 19.4.2
    see https://linkerd.io/checks/#l5d-data-plane-version for hints
‼ data plane and cli versions match
    linkerd/linkerd-identity-56867874c6-8jh29 running edge-19.4.1 but cli running edge-19.4.2
    see https://linkerd.io/checks/#l5d-data-plane-cli-version for hints

Status check results are √
```

### Environment

- Kubernetes Version: 1.12.5
- Cluster Environment: AKS
- Host OS:
- Linkerd version: edge-19.4.2

### Additional context

I previously upgraded this cluster from edge-19.3.3 to edge-19.4.1 using ```linkerd upgrade --ha --proxy-auto-inject > install.yaml```

**Comment by olix0r** [comment]
Thanks @jon-walton. Can you share the output of `grep -vF key.pem install.yaml` ? (i.e. please don't send us the private key ;) in a gist?

**Comment by olix0r** [comment]
This sounds potentially related to https://github.com/linkerd/linkerd2/issues/2660

**Comment by jon-walton** [comment]
@olix0r sure > https://gist.github.com/jon-walton/5a4f3c62cb4c176e678b413f7426f16e

**Comment by olix0r** [comment]
Thanks @jon-walton 

This is fixed by https://github.com/linkerd/linkerd2/pull/2662.

**Comment by jon-walton** [comment]
confirmed fixed in edge-19.4.3 🎉
